api/preprint_providers/views.py

```diff
@@ -75,13 +79,24 @@ class PreprintProviderList(JSONAPIBaseView, generics.ListAPIView, ODMFilterMixin
 
     ordering = ('name', )
 
-    # implement ODMFilterMixin
-    def get_default_odm_query(self):
-        return None
+    # implement DjangoFilterMixin
+    def get_default_django_query(self):
+        # TODO something more consistent: a filter[] param, or another endpoint (maybe under /users/me)
+        permission = self.request.GET.get('with_reviews_permission')
+        if permission is not None:
+            if permission not in reviews_permissions.PERMISSIONS:
+                raise InvalidQueryStringError('Invalid reviews permission', 'with_reviews_permission')
+            auth = get_user_auth(self.request)
+            auth_user = getattr(auth, 'user', None)
+            if not auth_user:
+                raise PermissionDenied()
+            return Q(id__in=PreprintProvider.review_providers_for_user(auth_user, permission))
+
+        return Q()
```

Comment:
Switching from ODM to Django mixins -- `PreprintProvider.find(None)` returns everything, but `PreprintProvider.objects.filter(None)` errors.